devel/ci/integration/tests/utils.py

```diff
@@ -0,0 +1,63 @@
+import os
+import subprocess
+import tempfile
+from contextlib import contextmanager
+
+import requests
+
+
+def get_db_dump(url):
+    filename = os.path.basename(url)
+    if filename.endswith(".xz"):
+        filename = filename[:-3]
+    filepath = os.path.join("devel", "ci", "integration", "dumps", filename)
+    if os.path.exists(filepath):
+        return filepath
+    response = requests.get(url)
+    if url.endswith(".xz"):
+        compressed_filepath = "{}.xz".format(filepath)
+        with open(compressed_filepath, "wb") as fd:
+            for chunk in response.iter_content(chunk_size=128):
+                fd.write(chunk)
+        subprocess.call(["xz", "-d", compressed_filepath])
+    return filepath
+
+
+def make_db_and_user(db_container, name, dump_url=None):
+    # Prepare the database
+    db_container.execute(
+        ["psql", "-q", "-U", "postgres", "-c", "CREATE USER {} CREATEDB;".format(name)]
+    )
+    db_container.execute(
+        [
+            "psql", "-q", "-U", "postgres", "-c",
+            (
+                "CREATE DATABASE {} WITH TEMPLATE = template0 ENCODING = 'UTF8' "
+                "LC_COLLATE = 'en_US.UTF-8' LC_CTYPE = 'en_US.UTF-8';"
+            ).format(name),
+        ]
+    )
+    if dump_url:
+        db_dump = get_db_dump(dump_url)
+        db_container.copy_to(db_dump, "/tmp/database.dump")
+        db_container.execute(["psql", "-q", "-U", name, "-f", "/tmp/database.dump"])
+        db_container.execute(["rm", "/tmp/database.dump"])
+
+
+@contextmanager
+def read_file(container, path, binary=False):
+    with tempfile.TemporaryDirectory() as tempdir:
+        destfile = os.path.join(tempdir, "file")
+        container.copy_from(path, destfile)
+        mode = "rb" if binary else "r"
+        with open(destfile, mode) as fh:
+            yield fh
+
+
+@contextmanager
+def edit_file(container, path):
```

Comment:
For test utility functions, let's add docblocks that explain their purpose, arguments, argument types, and return types.